### PR TITLE
69 fix release bugs

### LINKED ISSUE #69: Fixes to Release 0.7
- [x] When creating a new organization, the old organization is deleted. Create a test to verify that the founder can create multiple organizations. Also, verify that MemberOf can hold more than one organization. 
- [x] Vision in organization needs to consider with existing vision document. This is to be resolved with https://github.com/UniversalDot/universal-dot-node/issues/68
- [x] After tasks has been accepted and removed from storage, it should be cleared from the DAO task list as well.

### PR DESCRIPTION
**Pull request by f-gate**
fixes the bug where 1 person cannot create multiple organisations because the one before is removed.

Also replaces a Vec for a BoundedVec.

### CODE REVIEW

**Comment by stojanov-igor** [comment]
We found more bugs: 

- [x] `memberOF` always returns one organization. Try creating multiple orgs and add tests to verify it works.
- [x] When transferring ownership make sure you update `MemberOF` and `Members` storage items. Add tests also.

**Comment by stojanov-igor** [comment]
Hi, 

A milestone evaluator found one test failing. See the output below. 

- [x] Reproduce the test and fix it. 

`---- benchmarking::benchmark_tests::test_benchmarks stdout ----
failing benchmark tests:
create_task: "InvalidOrganization"
thread 'benchmarking::benchmark_tests::test_benchmarks' panicked at 'index out of bounds: the len is 0 but the index is 0', pallets/task/src/benchmarking.rs:103:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
update_task: Any { .. }
thread 'benchmarking::benchmark_tests::test_benchmarks' panicked at 'index out of bounds: the len is 0 but the index is 0', pallets/task/src/benchmarking.rs:135:25
start_task: Any { .. }
thread 'benchmarking::benchmark_tests::test_benchmarks' panicked at 'index out of bounds: the len is 0 but the index is 0', pallets/task/src/benchmarking.rs:163:25
remove_task: Any { .. }
thread 'benchmarking::benchmark_tests::test_benchmarks' panicked at 'index out of bounds: the len is 0 but the index is 0', pallets/task/src/benchmarking.rs:191:25
complete_task: Any { .. }
thread 'benchmarking::benchmark_tests::test_benchmarks' panicked at 'index out of bounds: the len is 0 but the index is 0', pallets/task/src/benchmarking.rs:220:25
accept_task: Any { .. }
thread 'benchmarking::benchmark_tests::test_benchmarks' panicked at 'index out of bounds: the len is 0 but the index is 0', pallets/task/src/benchmarking.rs:252:25
reject_task: Any { .. }
thread 'benchmarking::benchmark_tests::test_benchmarks' panicked at 'assertion failed: !anything_failed', pallets/task/src/benchmarking.rs:265:1`

**Comment by f-gate** [comment]
i believe this is fixed in [46-expired tasks ](https://github.com/UniversalDot/universal-dot-node/pull/94). as Organisation Hashes were parsed in that dont exist the initial error of "InvalidOrganisation" means that the task cannot be created and therefore the indexing to get the tasks are out of bounds.

It is contained to benchmarking.

File: pallets/dao/src/lib.rs [hunk 1]
```diff
@@ -577,14 +602,19 @@ pub mod pallet {
...
 			// Update Organization Members
 			<Members<T>>::insert(org_id, members);
 
 			// Find current organizations and remove org_id from MemberOf user
 			let mut current_organizations = <Pallet<T>>::member_of(&account);
 			ensure!(current_organizations.iter().any(|a| *a == org_id), Error::<T>::InvalidOrganization);
-			current_organizations.retain(|a| *a != org_id);
+
 			// Update MemberOf
+			current_organizations = current_organizations.into_iter().filter(|a| *a !=
```
Comment (stojanov-igor) [inline pallets/dao/src/lib.rs:614]: isnit this equivalent to `current_organizations.retain(|a| *a != org_id);` which was removed?

File: pallets/dao/src/tests.rs [hunk 2]
```diff
@@ -599,4 +599,44 @@ fn owner_can_not_transfer_ownership_to_itself() {
...
+		let _org_id_2 = create_organization_2();
+
+		// Assert ALICE is a member of 2 Daos + 2 exist
+		assert!(Dao::member_of(*ALICE).len() == 2);
+		assert!(Dao::organization_count() == 2);
+	});
+}
+
+#[test]
+fn test_members_are_mutated_on_ownership_transfer() {
```
Comment (stojanov-igor) [inline pallets/dao/src/tests.rs:618]: Nice!

**Review by stojanov-igor — APPROVED** [review]
Nice work..

I did some minor fixes. 

@f-gate For every new piece of code we add, we should add inline comments that explain what the code is doing.

File: pallets/dao/src/lib.rs (hunk 1 above)
Comment (f-gate) [reply pallets/dao/src/lib.rs:614]: After some reading, retain should be used if you dont mind consuming the variable. My benchmarking didnt have it any quicker but it doesnt copy the vec like filter does. :). Lazy from me.

**Pull request merged**